#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.13970402465979537 0.9789225910205963 0.14897431417320728 4.62694260888857
  position -5.406883466768003 2.116734193190879 0.04039265975640029
}
RectangleArena {
  rotation 0 1 0 4.6928e-06
  floorSize 3.4 3.4
  floorTileSize 0.4 0.4
  floorAppearance Parquetry {
  }
  wallThickness 0.2
  wallHeight 0.2
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
DEF ROBOT0 Robot {
  translation 0.010000002427230877 0.039483576832102844 1.08
  rotation 3.862038331050773e-08 6.523870273143482e-09 -0.9999999999999993 6.067485953280969e-08
  children [
    DEF EMITTER Emitter {
    }
    DEF RECEIVER Receiver {
    }
    DEF HINGE_COVER Solid {
      translation 0 0.03 -3.469446951953614e-18
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.6509803921568628 1
          }
          geometry Box {
            size 0.030000000000000002 0.019999999999999997 0.05
          }
        }
      ]
      name "hingeCover"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 6.543046716983263e-17
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 2.909064282420552e-14 -2.993457139500296e-14 1 1.5708000000006888
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position -6.221264785464671e-17
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 2.657571964934993e-14 -2.7733656753129075e-14 1 1.570800000000685
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 1.1066660321523367e-16
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 4.6335813229031846e-14 -4.6649840222151485e-14 1 1.5708000000006888
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position -1.6403220341991584e-16
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -1.4161861526089543e-14 1.5300188765441183e-14 1 1.570800000000689
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF POLE HingeJoint {
      jointParameters HingeJointParameters {
        position -3.2190502794938286e-12
        anchor 0 0.03000000000047226 0
        minStop -1.3
        maxStop 1.3
      }
      device [
        DEF POLE_POS_SENSOR PositionSensor {
          name "polePosSensor"
        }
      ]
      endPoint Solid {
        translation -1.8379558561220447e-06 0.5299968874104775 -6.458533960745036e-11
        rotation -6.45855341688599e-11 6.45863989111891e-11 1 1.5708000033841265
        children [
          DEF POLE_ENDPOINT_0 Solid {
            translation 0.5000000000000002 0 0
            scale 0.1 0.1 0.1
            children [
              DEF BALL Shape {
                appearance PBRAppearance {
                  baseColor 0.03137254901960784 0.047058823529411764 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.1
                }
              }
            ]
            boundingObject USE BALL
            physics Physics {
              density -1
              mass 0.02
            }
          }
          DEF POLE Shape {
            appearance PBRAppearance {
              baseColor 1 0.1568627450980392 0.1568627450980392
              roughness 1
              metalness 0
            }
            geometry Box {
              size 1 0.020000000000000004 0.019999999999999997
            }
          }
        ]
        name "pole"
        boundingObject USE POLE
        physics Physics {
          density -1
          mass 0.2
        }
      }
    }
  ]
  name "ROBOT0"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 2
  }
  controller "test_robot"
}
DEF ROBOT1 Robot {
  translation 0.010000002427230904 0.03948357683210284 -0.99
  rotation 3.696711334805196e-08 2.6569309114573028e-09 -0.9999999999999993 6.067485986022764e-08
  children [
    DEF EMITTER Emitter {
      channel 1
    }
    DEF RECEIVER Receiver {
      channel 1
    }
    DEF HINGE_COVER Solid {
      translation 0 0.03 -3.469446951953614e-18
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.6509803921568628 1
          }
          geometry Box {
            size 0.030000000000000002 0.019999999999999997 0.05
          }
        }
      ]
      name "hingeCover"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position -3.304848396029972e-17
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 2.712107103625866e-14 -2.7964992372400403e-14 1 1.5708000000006888
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position -1.7824701330397307e-16
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 2.425504086470964e-14 -2.5412969444142682e-14 1 1.570800000000685
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 4.0403759658600144e-17
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 4.4930561519687186e-14 -4.524458335101898e-14 1 1.5708000000006888
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position -2.7421646697179155e-16
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -1.6365538702552456e-14 1.750387403647872e-14 1 1.570800000000689
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF POLE HingeJoint {
      jointParameters HingeJointParameters {
        position -3.2218524345923727e-12
        anchor 0 0.03000000000047226 0
        minStop -1.3
        maxStop 1.3
      }
      device [
        DEF POLE_POS_SENSOR PositionSensor {
          name "polePosSensor"
        }
      ]
      endPoint Solid {
        translation -1.8379558561220447e-06 0.5299968874104775 -6.46413823605429e-11
        rotation -6.46415770647838e-11 6.464244201315995e-11 1 1.5708000033841265
        children [
          DEF POLE_ENDPOINT_1 Solid {
            translation 0.5000000000000002 0 0
            scale 0.1 0.1 0.1
            children [
              DEF BALL Shape {
                appearance PBRAppearance {
                  baseColor 0.03137254901960784 0.047058823529411764 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.1
                }
              }
            ]
            boundingObject USE BALL
            physics Physics {
              density -1
              mass 0.02
            }
          }
          DEF POLE Shape {
            appearance PBRAppearance {
              baseColor 1 0.1568627450980392 0.1568627450980392
              roughness 1
              metalness 0
            }
            geometry Box {
              size 1 0.020000000000000004 0.019999999999999997
            }
          }
        ]
        name "pole"
        boundingObject USE POLE
        physics Physics {
          density -1
          mass 0.2
        }
      }
    }
  ]
  name "ROBOT1"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 2
  }
  controller "test_robot"
}
Robot {
  children [
    DEF receiver0 Receiver {
      name "receiver0"
    }
    DEF receiver1 Receiver {
      name "receiver1"
      channel 1
    }
    DEF emitter0 Emitter {
      name "emitter0"
    }
    DEF emitter1 Emitter {
      name "emitter1"
      channel 1
    }
  ]
  name "supervisor"
  controller "test_supervisor"
  supervisor TRUE
}
